### PR TITLE
Feature/xtg 298 CRUD UserRole and Capability

### PR DESCRIPTION
**Pull request by fredericobraga85**
ticket: https://x-team-internal.atlassian.net/browse/XTG-298

add CRUD for UserRole
add CRUD for Capability
add tests
add test coverage logic

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,11 +8,13 @@
     "start:noseed": "node dist/src/index.js -- noseed",
     "dev": "npm install --frozen-lockfile --production=false && npm run build && npm run prepare-db && concurrently \"tsc --watch\" nodemon",
     "test": "cross-env NODE_ENV=test ENV=test npm run test:all",
+    "test:coverage": "cross-env NODE_ENV=test ENV=test npm run test:all:coverage",
```
Comment (fredericobraga85) [inline package.json:11]: adding a script for test coverage. Still a lot of work to do:

File: src/migrations/20220628100710-edit-UserRoleCapability-table-constraint.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import type { QueryInterface, Sequelize } from 'sequelize';
+
+interface SequelizeContext {
+  context: {
+    queryInterface: QueryInterface;
+    Sequelize: Sequelize;
+  };
+}
+module.exports = {
+  async up({ context: { queryInterface } }: SequelizeContext) {
```
Comment (ramirlm) [inline src/migrations/20220628100710-edit-UserRoleCapability-table-constraint.ts:11]: Same Q for this one...

File: src/migrations/20220628090633-edit-Capability-table-constraint.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          type: DataTypes.TEXT,
+          unique: true,
+        },
+        { transaction }
+      );
+    });
+  },
+
+  async down({ context: { queryInterface } }: SequelizeContext) {
+    return queryInterface.sequelize.transaction(async (transaction) => {
```
[the remark this answers is not in the document]
Comment (fredericobraga85) [reply src/migrations/20220628090633-edit-Capability-table-constraint.ts:27]: they are are not exactly the same . one is unique true and the other  unique: false

File: src/migrations/20220628100710-edit-UserRoleCapability-table-constraint.ts (hunk 2 above)
Comment (fredericobraga85) [reply src/migrations/20220628100710-edit-UserRoleCapability-table-constraint.ts:11]: up handles the onDelete and onUpdate with CASCADE while the down handles onDelete and onUpdate with Restrict

**Pull request merged**